### ISSUE TITLE
[Bug]:  Dnsmasq DHCP integrations returns wrong data

### ISSUE THREAD

**Issue by rhoriguchi** [netdata/netdata#18392]
### Bug description

The module is showing me it detected 5 dhcp ranges but in the output, it will show me 0. Also, it will only show me 3 hosts which are all static ones and one is not even online.

### Expected behavior

It should show 5 ranges and 9 hosts. And calculate allocations correctly.

### Steps to reproduce

1. Download these 3 files to `/tmp/tmp.Y33g18NNNx` or any other temp dir. Just make sure to adjust values in `dnsmasq_dhcp.conf`. Remove `.txt` from file extensions.
   -  [dnsmasq.conf](https://github.com/user-attachments/files/16698121/dnsmasq.conf.txt)
   -  [dnsmasq.leases](https://github.com/user-attachments/files/16698125/dnsmasq.leases.txt)
   - [dnsmasq_dhcp.conf](https://github.com/user-attachments/files/16698131/dnsmasq_dhcp.conf.txt)
2. Run `sudo -u netdata -s ./go.d.plugin -d -m dnsmasq_dhcp -c /tmp/tmp.Y33g18NNNx`



### Installation method

from source

### System info

```shell
> uname -a; grep -HvE "^#|URL" /etc/*release
Linux XXLPitu-Router 6.10.5 #1-NixOS SMP PREEMPT_DYNAMIC Wed Aug 14 13:34:38 UTC 2024 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_CODENAME=vicuna
/etc/lsb-release:DISTRIB_DESCRIPTION="NixOS 24.11 (Vicuna)"
/etc/lsb-release:DISTRIB_ID=nixos
/etc/lsb-release:DISTRIB_RELEASE="24.11"
/etc/lsb-release:LSB_VERSION="24.11 (Vicuna)"
/etc/os-release:ANSI_COLOR="1;34"
/etc/os-release:BUILD_ID="24.11.20240818.8a33541"
/etc/os-release:ID=nixos
/etc/os-release:IMAGE_ID=""
/etc/os-release:IMAGE_VERSION=""
/etc/os-release:LOGO="nix-snowflake"
/etc/os-release:NAME=NixOS
/etc/os-release:PRETTY_NAME="NixOS 24.11 (Vicuna)"
/etc/os-release:VERSION="24.11 (Vicuna)"
/etc/os-release:VERSION_CODENAME=vicuna
/etc/os-release:VERSION_ID="24.11"
```


### Netdata build info

```shell
Packaging:
    Netdata Version ____________________________________________ : v1.46.1
    Installation Type __________________________________________ : unknown
    Package Architecture _______________________________________ : unknown
    Package Distro _____________________________________________ : unknown
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /etc/netdata
    Stock Configurations _______________________________________ : /nix/store/d08j7jf9fcg9h67amksqa8qyj14bngwa-netdata-1.46.1/share/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /var/cache/netdata
    Permanent Databases ________________________________________ : /var/lib/netdata
    Plugins ____________________________________________________ : /nix/store/d08j7jf9fcg9h67amksqa8qyj14bngwa-netdata-1.46.1/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /nix/store/d08j7jf9fcg9h67amksqa8qyj14bngwa-netdata-1.46.1/share/netdata/web
    Log Files __________________________________________________ : /var/log/netdata
    Lock Files _________________________________________________ : /var/lib/netdata/lock
    Home _______________________________________________________ : /var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 6.10.5
    Operating System ___________________________________________ : NixOS
    Operating System ID ________________________________________ : nixos
    Operating System ID Like ___________________________________ : unknown
    Operating System Version ___________________________________ : 24.11 (Vicuna)
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : /etc/os-release
Hardware:
    CPU Cores __________________________________________________ : 8
    CPU Frequency ______________________________________________ : 1800000000
    RAM Bytes __________________________________________________ : 16540794880
    Disk Capacity ______________________________________________ : 125099311104
    CPU Architecture ___________________________________________ : x86_64
    Virtualization Technology __________________________________ : none
    Virtualization Detection ___________________________________ : systemd-detect-virt
Container:
    Container __________________________________________________ : none
    Container Detection ________________________________________ : systemd-detect-virt
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : NO (unavailable)
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip brotli)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine (compression) _____________________________________ : YES (zstd lz4)
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : NO
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : YES
    protobuf (platform-neutral data serialization protocol) ____ : NO
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : YES
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : NO
    ebpf (monitor system calls) ________________________________ : NO
    freeipmi (monitor enterprise server H/W) ___________________ : YES
    nfacct (gather netfilter accounting) _______________________ : YES
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : NO
    Xen VBD Error Tracking _____________________________________ : NO
    Logs Management ____________________________________________ : YES
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : NO
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : NO
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```


### Additional info

Log output:

```
DBG godplugin/main.go:142 plugin: name=go.d, version=v1.46.1 component=agent
DBG godplugin/main.go:144 current user: name=rhoriguchi, uid=1000 component=agent
INF godplugin/main.go:148 env HTTP_PROXY '', HTTPS_PROXY '' component=agent
INF agent/agent.go:145 instance is started component=agent
INF agent/setup.go:23 loading config file component=agent
DBG agent/setup.go:31 looking for 'go.d.conf' in [/tmp/tmp.Y33g18NNNx] component=agent
WRN agent/setup.go:35 couldn't find config, will use defaults component=agent
INF agent/agent.go:149 using config: enabled 'true', default_run 'true', max_procs '0' component=agent
INF agent/setup.go:50 loading modules component=agent
INF agent/setup.go:73 enabled/registered modules: 1/92 component=agent
INF agent/setup.go:79 building discovery config component=agent
DBG agent/setup.go:126 looking for 'dnsmasq_dhcp.conf' in [/tmp/tmp.Y33g18NNNx] component=agent
DBG agent/setup.go:142 found '/tmp/tmp.Y33g18NNNx/dnsmasq_dhcp.conf component=agent
INF agent/setup.go:147 dummy/read/watch paths: 0/1/0 component=agent
INF discovery/manager.go:116 registered discoverers: [file discovery: [file reader] service discovery] component="discovery manager"
DBG agent/setup.go:165 looking for 'vnodes/' in [/tmp/tmp.Y33g18NNNx] component=agent
INF discovery/manager.go:61 instance is started component="discovery manager"
INF functions/manager.go:50 instance is started component="functions manager"
INF file/discovery.go:69 instance is started component=discovery discoverer=file
INF file/read.go:48 instance is started component=discovery discoverer=file
INF jobmgr/manager.go:87 instance is started component="job manager"
DBG functions/ext.go:15 registering function 'config' component="functions manager"
CONFIG go.d:collector:dnsmasq_dhcp create accepted template /collectors/jobs internal 'internal' 'add schema enable disable test userconfig' 0x0000 0x0000

INF sd/sd.go:66 instance is started component="service discovery"
INF file/read.go:49 instance is stopped component=discovery discoverer=file
DBG jobmgr/manager.go:122 received configs: 1/+1/-0 ('/tmp/tmp.Y33g18NNNx/dnsmasq_dhcp.conf') component="job manager"
CONFIG go.d:collector:dnsmasq_dhcp:dnsmasq_dhcp create accepted job /collectors/jobs stock 'discoverer=file_reader,file=/tmp/tmp.Y33g18NNNx/dnsmasq_dhcp.conf' 'schema get enable disable update restart test userconfig' 0x0000 0x0000

DBG jobmgr/manager.go:293 creating dnsmasq_dhcp[dnsmasq_dhcp] job, config: map[__provider__:file reader __source__:discoverer=file_reader,file=/tmp/tmp.Y33g18NNNx/dnsmasq_dhcp.conf __source_type__:stock autodetection_retry:0 conf_path:/tmp/
tmp.Y33g18NNNx/dnsmasq.conf leases_path:/tmp/tmp.Y33g18NNNx/dnsmasq.leases module:dnsmasq_dhcp name:dnsmasq_dhcp priority:70000 update_every:1] component="job manager"
DBG dnsmasq_dhcp/parse_configuration.go:33 looking in '/tmp/tmp.Y33g18NNNx/dnsmasq.conf' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:36 found dhcp-range 'enp3s0,172.16.1.2,172.16.1.254,1h' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:36 found dhcp-range 'enp2s0,192.168.1.2,192.168.1.254,1h' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:36 found dhcp-range 'enp2s0.1,192.168.1.2,192.168.1.254,1h' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:36 found dhcp-range 'enp2s0.2,192.168.2.2,192.168.2.254,1h' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:36 found dhcp-range 'enp2s0.3,192.168.3.2,192.168.3.254,1h' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:36 found dhcp-range 'enp2s0.100,192.168.100.2,192.168.100.254,1h' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:65 looking in '/tmp/tmp.Y33g18NNNx/dnsmasq.conf' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:68 found dhcp-host '3d:a4:b9:5f:e7:05,device-1,192.168.1.2' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:80 adding dhcp-host '192.168.1.2' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:68 found dhcp-host '26:f9:f4:ee:f2:51,device-2,192.168.2.2' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:80 adding dhcp-host '192.168.2.2' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:68 found dhcp-host 'cc:83:4b:77:2d:80,192.168.3.254' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/parse_configuration.go:80 adding dhcp-host '192.168.3.254' collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/collect.go:49 leases db file modification time has changed, reading it collector=dnsmasq_dhcp job=dnsmasq_dhcp
INF module/job.go:244 check success collector=dnsmasq_dhcp job=dnsmasq_dhcp
CONFIG go.d:collector:dnsmasq_dhcp:dnsmasq_dhcp status running
                                                                                                                                                                                                                                 22:24:58 [8/58]
INF module/job.go:266 started, data collection interval 1s collector=dnsmasq_dhcp job=dnsmasq_dhcp
DBG dnsmasq_dhcp/collect.go:45 lease database file modification time has not changed, old data is returned collector=dnsmasq_dhcp job=dnsmasq_dhcp
CHART 'netdata.execution_time_of_dnsmasq_dhcp' '' 'Execution time' 'ms' 'go.d' 'netdata.go_plugin_execution_time' 'line' '145000' '1' '' 'go.d' 'dnsmasq_dhcp'
CLABEL '_collect_job' 'dnsmasq_dhcp' '1'
CLABEL_COMMIT
DIMENSION 'time' '' 'absolute' '1' '1' ''

CHART 'dnsmasq_dhcp.dhcp_ranges' '' 'Number of DHCP Ranges' 'ranges' 'dhcp ranges' 'dnsmasq_dhcp.dhcp_ranges' 'stacked' '70002' '1' '' 'go.d' 'dnsmasq_dhcp'
CLABEL '_collect_job' 'dnsmasq_dhcp' '1'
CLABEL_COMMIT
DIMENSION 'ipv4' 'ipv4' 'absolute' '1' '1' ''
DIMENSION 'ipv6' 'ipv6' 'absolute' '1' '1' ''

BEGIN 'dnsmasq_dhcp.dhcp_ranges'
SET 'ipv4' = 0
SET 'ipv6' = 0
END

CHART 'dnsmasq_dhcp.dhcp_hosts' '' 'Number of DHCP Hosts' 'hosts' 'dhcp hosts' 'dnsmasq_dhcp.dhcp_host' 'stacked' '70003' '1' '' 'go.d' 'dnsmasq_dhcp'
CLABEL '_collect_job' 'dnsmasq_dhcp' '1'
CLABEL_COMMIT
DIMENSION 'ipv4' 'ipv4' 'absolute' '1' '1' ''
DIMENSION 'ipv6' 'ipv6' 'absolute' '1' '1' ''

BEGIN 'dnsmasq_dhcp.dhcp_hosts'
SET 'ipv4' = 3
SET 'ipv6' = 0
END

BEGIN 'netdata.execution_time_of_dnsmasq_dhcp'
SET 'time' = 0
END

DBG dnsmasq_dhcp/collect.go:45 lease database file modification time has not changed, old data is returned collector=dnsmasq_dhcp job=dnsmasq_dhcp
BEGIN 'dnsmasq_dhcp.dhcp_ranges' 1000078
SET 'ipv4' = 0
SET 'ipv6' = 0
END

BEGIN 'dnsmasq_dhcp.dhcp_hosts' 1000078
SET 'ipv4' = 3
SET 'ipv6' = 0
END

BEGIN 'netdata.execution_time_of_dnsmasq_dhcp' 1000078
SET 'time' = 0
END

^CINF agent/agent.go:113 received interrupt signal (2). Terminating... component=agent
INF file/discovery.go:70 instance is stopped component=discovery discoverer=file
INF sd/sd.go:67 instance is stopped component="service discovery"
INF module/job.go:267 stopped collector=dnsmasq_dhcp job=dnsmasq_dhcp
INF functions/manager.go:51 instance is stopped component="functions manager"
INF jobmgr/manager.go:88 instance is stopped component="job manager"
INF discovery/manager.go:62 instance is stopped component="discovery manager"
INF agent/agent.go:146 instance is stopped component=agent
```

**Comment by ilyam8** [comment]
@rhoriguchi parsing of dhcp-ranges fixed in #18392

>  Also, it will only show me 3 hosts which are all static ones and one is not even online.

^^The number of `dhcp-host` in your config, clients are the leases ("DHCP Range Allocated Leases" chart).

**Comment by ilyam8** [comment]
<details>
<summary>Also updated UI</summary>

<img width="1666" alt="Screenshot 2024-08-22 at 13 46 07" src="https://github.com/user-attachments/assets/9c6295c4-db4a-48b7-8e00-644981c2bb56">


</details>
